kit/controls/src/UIShowMoreButton.tsx

```diff
@@ -0,0 +1,172 @@
+import { UILayoutConstant } from '@tonlabs/uikit.layout';
+import { uiLocalized } from '@tonlabs/localization';
+import {
+    UILabel,
+    UILabelColors,
+    UILabelRoles,
+    ColorVariants,
+    useTheme,
+    Theme,
+    TypographyVariants,
+    makeStyles,
+} from '@tonlabs/uikit.themes';
+
+import { TouchableOpacity } from './TouchableOpacity';
+import { UIConstant } from './constants';
+
+if (Platform.OS === 'android' && UIManager.setLayoutAnimationEnabledExperimental) {
+    UIManager.setLayoutAnimationEnabledExperimental(true);
+}
+
+export enum UIShowMoreButtonHeight {
+    Medium = 'Medium',
+    Small = 'Small',
+}
+
+export type UIShowMoreButtonProps = {
+    label?: string;
+    progress: boolean;
+    height?: UIShowMoreButtonHeight;
+    onPress?: () => void;
+    background?: boolean;
+    testID?: string;
+};
+
+function getLabelRole(showMoreButtonHeight: UIShowMoreButtonHeight): TypographyVariants {
+    switch (showMoreButtonHeight) {
+        case UIShowMoreButtonHeight.Small:
+            return UILabelRoles.NarrowHeadlineFootnote;
+        case UIShowMoreButtonHeight.Medium:
+        default:
+            return UILabelRoles.Action;
+    }
+}
+
+function getIndicatorSize(showMoreButtonHeight: UIShowMoreButtonHeight): number {
+    switch (showMoreButtonHeight) {
+        case UIShowMoreButtonHeight.Small:
+            return UIConstant.showMoreButtonIndicatorSizeSmall;
+        case UIShowMoreButtonHeight.Medium:
+        default:
+            return UIConstant.showMoreButtonIndicatorSizeMedium;
+    }
+}
+
+function getSize(showMoreButtonHeight: UIShowMoreButtonHeight): number {
+    switch (showMoreButtonHeight) {
+        case UIShowMoreButtonHeight.Small:
+            return UIConstant.showMoreButtonSizeSmall;
+        case UIShowMoreButtonHeight.Medium:
+        default:
+            return UIConstant.showMoreButtonSizeMedium;
+    }
+}
+
+const Indicator = React.memo(
+    ({
+        showMoreButtonHeight,
+        color,
+    }: {
+        showMoreButtonHeight: UIShowMoreButtonHeight;
+        color: string;
+    }) => {
+        return (
+            <MaterialIndicator
+                style={{ position: 'absolute' }}
+                color={color}
+                size={getIndicatorSize(showMoreButtonHeight)}
+            />
+        );
+    },
+);
+
+export const UIShowMoreButton: React.FunctionComponent<UIShowMoreButtonProps> = ({
+    label = uiLocalized.ShowMore,
+    progress = false,
+    height = UIShowMoreButtonHeight.Medium,
+    onPress: onPressProp,
+    background = true,
+    testID,
+}: UIShowMoreButtonProps) => {
+    const theme = useTheme();
+    const styles = useStyles(theme, background, height, progress);
+
+    const onPress = React.useCallback(() => {
+        if (onPressProp) {
+            onPressProp();
+        }
+    }, [onPressProp]);
+
+    React.useLayoutEffect(() => {
+        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
+    }, [progress]);
```

Comment:
I wonder how does this animation look like?)

Btw, does "weak" Android can handle it without any problems?

Reply:
android and iOS work perfectly. Animation doesn't work on the web